src/curl.cpp

```diff
@@ -343,6 +343,23 @@ static const long S3FSCURL_RESPONSECODE_NOTSET      = -1;
 static const long S3FSCURL_RESPONSECODE_FATAL_ERROR = -2;
 static const int  S3FSCURL_PERFORM_RESULT_NOTSET    = 1;
 
+// [NOTE]
+// This is the search path array for the mime.types file, ordered
+// by priority.
+// On most operating systems(Debian, Ubuntu, CentOS, openSUSE and
+// etc.), /etc/mime.types is expected. However, this file is almost
+// absent on macOS, but /etc/apache2/mime.types exists instead of
+// it. So, when s3fs loads the mime.types file, it tries to find
+// mime file in the order of this array.
```

Comment:
I wonder if there could be some corner cases on GNU/Linux where `/etc/apache2/mime.types` exists, but `/etc/mime.types` does not.

If such situation is possible, then this would be wrong, as s3fs-fuse would be depending on Apache instead of depending on the system package that provides `/etc/mime.types`. So if at some point Apache is removed, s3fs-fuse will stop working.

Personally, I'd look for for `/etc/mime.types for all OS, and if it doesn't exist only for `/etc/apache2/mime.types` on macOS.

Reply:
@juliogonzalez Thanks for your help.
I could understand your concern. (please point out if wrong)
Certainly, It is not good that s3fs depends on apache.
I'll try to change that it only treat macOS as special(apache preinstalled OS).

The final specification will be changed as follows:
For macOS, s3fs try to detect /etc/apache2/mime.types first, and if it doesn't exist, look for /etc/mime.types.
For non-macOS, it try to detect only /etc/mime.types.
If any file cannot be found, both s3fs will fail to start.
However, if user specify the `mime` option, s3fs will try to detect only the file specified by that option.

I try making the code changes again, please wait a few days.

Reply:
> For macOS, s3fs try to detect /etc/apache2/mime.types first, and if it doesn't exist, look for /etc/mime.types.
> For non-macOS, it try to detect only /etc/mime.types.

Wouldn't it be easier:

1. s3fs try to detect `/etc/mime.types` regardless of the OS
2. Else s3fs try to detect `/etc/apache2/mime.types` if OS is macOS
3. Else exit with an en error.

?

Benefit: you treat `/etc/apache2/mime.types` as an exception at macOS, and only if the standard file doesn't exist. If the standard file exists on past or future versions, then it will be used.

Reply:
Thanks, I will up codes with squashing soon.